### PR TITLE
feat: improve and add database setting in asset pages

### PR DESCRIPTION
**Pull request by leo220yuyaodog**

### CODE REVIEW

File: web/src/Conf.js [hunk 1]
```diff
@@ -19,6 +19,7 @@ export const AuthConfig = {
   appName: "app-casvisor",
   organizationName: "casbin",
   redirectPath: "/callback",
+  dbgateUrl: "http://localhost:5001",
```
Comment (hsluoyz) [inline web/src/Conf.js]: It's not AuthConfig

File: web/src/Setting.js [hunk 2]
```diff
@@ -326,3 +326,12 @@ export function GetIdFromObject(obj) {
   }
   return `${obj.owner}/${obj.name}`;
 }
+
+export const DataBaseTypes = [
+  {label: "MySQL", value: "mysql"},
```
Comment (hsluoyz) [inline web/src/Setting.js]: value and label should be the same, we store what users see in UI

File: web/src/AssetListPage.js [hunk 3]
```diff
@@ -121,32 +120,67 @@ class AssetListPage extends BaseListPage {
...
+        key: "databaseType",
+        width: "150px",
         sorter: true,
         filterMultiple: false,
         filters: [
-          {text: "RDP", value: "RDP"},
-          {text: "VNC", value: "VNC"},
-          {text: "SSH", value: "SSH"},
-          {text: "", value: "-"},
+          {text: "MySQL", value: "mysql"},
```
Comment (hsluoyz) [inline web/src/AssetListPage.js]: Generate filter from Setting.DataBaseTypes ?

File: web/src/Setting.js [hunk 4]
```diff
@@ -326,3 +326,12 @@ export function GetIdFromObject(obj) {
   }
   return `${obj.owner}/${obj.name}`;
 }
+
+export const DataBaseTypes = [
```
Comment (hsluoyz) [inline web/src/Setting.js]: DataBaseTypes -> getDatabaseTypes

File: web/src/AssetListPage.js [hunk 5]
```diff
@@ -121,32 +120,67 @@ class AssetListPage extends BaseListPage {
...
       {
         title: i18next.t("general:Protocol"),
         dataIndex: "protocol",
         key: "protocol",
-        width: "50px",
+        width: "100px",
+        sorter: true,
+        filterMultiple: false,
+        filters: [
+          {text: "RDP", value: "rdp"},
```
Comment (hsluoyz) [inline web/src/AssetListPage.js]: Also define Setting.getMachineTypes

File: web/src/AssetListPage.js [hunk 6]
```diff
@@ -294,9 +273,15 @@ class AssetListPage extends BaseListPage {
   fetch = (params = {}) => {
     let field = params.searchedColumn, value = params.searchText;
     const sortField = params.sortField, sortOrder = params.sortOrder;
-    if (params.type !== undefined && params.type !== null) {
-      field = "type";
-      value = params.type;
+    if (params.category !== undefined && params.category !== null) {
```
Comment (hsluoyz) [inline web/src/AssetListPage.js]: Use `if (params.category)`

File: web/src/AssetEditPage.js [hunk 7]
```diff
@@ -169,47 +195,83 @@ class AssetEditPage extends React.Component {
...
+        {
+          asset.category === "Machine" &&
+          <Row style={{marginTop: "20px"}} >
+            <Col style={{marginTop: "5px"}} span={(Setting.isMobile()) ? 22 : 2}>
+              {Setting.getLabel(i18next.t("general:Protocol"), i18next.t("general:Protocol - Tooltip"))} :
+            </Col>
+            <Col span={22} >
+              <Select virtual={false} style={{width: "100%"}} value={asset.protocol}
+                options={[
+                  {label: "RDP", value: "rdp"},
```
Comment (hsluoyz) [inline web/src/AssetEditPage.js]: Use Setting.getMachineTypes

File: web/src/AssetEditPage.js [hunk 8]
```diff
@@ -169,47 +195,83 @@ class AssetEditPage extends React.Component {
...
+                  {label: "VNC", value: "vnc"},
+                  {label: "SSH", value: "ssh"},
```
Comment (hsluoyz) [inline web/src/AssetEditPage.js]: Value also use label, VNC, RDP, etc.

File: web/src/AssetEditPage.js [hunk 9]
```diff
@@ -130,21 +126,51 @@ class AssetEditPage extends React.Component {
...
   }
 
-  getDefaultPort(protocol) {
-    if (protocol === "rdp") {
+  getDefaultPort(key) {
+    switch (key) {
+    case "rdp":
       return 3389;
-    } else if (protocol === "vnc") {
+    case "vnc":
```
Comment (hsluoyz) [inline web/src/AssetEditPage.js]: VNC, SSH, RDP

File: object/asset.go [hunk 10]
```diff
@@ -45,21 +45,29 @@ type Asset struct {
...
+	DisplayName string `xorm:"varchar(100)" json:"displayName"`
 	Ip          string `xorm:"varchar(100)" json:"ip"`
 	Port        int    `json:"port"`
 	Username    string `xorm:"varchar(100)" json:"username"`
-	Password    string `xorm:"varchar(100)" json:"password"`
-	Language    string `xorm:"varchar(100)" json:"language"`
+	Password    string `xorm:"varchar(200)" json:"password"`
+	Language    string `xorm:"varchar(20)" json:"language"`
+	Category    string `xorm:"varchar(20)" json:"category"`
 	Tag         string `xorm:"varchar(100)" json:"tag"`
```
Comment (hsluoyz) [inline object/asset.go]: Add Type

File: object/asset.go [hunk 11]
```diff
@@ -45,21 +45,29 @@ type Asset struct {
 	Name        string `xorm:"varchar(100) notnull pk" json:"name"`
 	CreatedTime string `xorm:"varchar(100)" json:"createdTime"`
 
-	Description string `xorm:"varchar(100)" json:"description"`
-	Protocol    string `xorm:"varchar(100)" json:"protocol"`
+	DisplayName string `xorm:"varchar(100)" json:"displayName"`
 	Ip          string `xorm:"varchar(100)" json:"ip"`
```
Comment (hsluoyz) [inline object/asset.go]: Endpoint

File: object/asset.go [hunk 12]
```diff
@@ -45,21 +45,29 @@ type Asset struct {
...
-	Language    string `xorm:"varchar(100)" json:"language"`
+	Password    string `xorm:"varchar(200)" json:"password"`
+	Language    string `xorm:"varchar(20)" json:"language"`
+	Category    string `xorm:"varchar(20)" json:"category"`
 	Tag         string `xorm:"varchar(100)" json:"tag"`
 	Os          string `xorm:"varchar(100)" json:"os"`
-	AutoQuery   bool   `json:"autoQuery"`
-	IsPermanent bool   `json:"isPermanent"`
 
+	Protocol        string       `xorm:"varchar(20)" json:"protocol"`
```
Comment (hsluoyz) [inline object/asset.go]: Protocol -> Type

File: object/asset.go [hunk 13]
```diff
@@ -45,21 +45,29 @@ type Asset struct {
...
+	AutoQuery       bool         `json:"autoQuery"`
+	IsPermanent     bool         `json:"isPermanent"`
 	EnableRemoteApp bool         `json:"enableRemoteApp"`
 	RemoteApps      []*RemoteApp `json:"remoteApps"`
 	Services        []*Service   `json:"services"`
+
+	Id              string `xorm:"varchar(100) index" json:"id"`
+	DatabaseType    string `xorm:"varchar(100)" json:"databaseType"`
```
Comment (hsluoyz) [inline object/asset.go]: DatabaseType -> Type

File: object/asset.go [hunk 14]
```diff
@@ -45,21 +45,29 @@ type Asset struct {
...
+	AuthType        string `xorm:"varchar(100)" json:"authType"`
+	DefaultDatabase string `xorm:"varchar(100)" json:"defaultDatabase"`
+	DatabaseUrl     string `xorm:"varchar(200)" json:"databaseUrl"`
```
Comment (hsluoyz) [inline object/asset.go]: What's DatabaseUrl and UseDatabaseUrl ?

**Review by hsluoyz — CHANGES_REQUESTED** [review]

File: object/asset.go (hunk 14 above)
Comment (leo220yuyaodog) [reply object/asset.go]: dbgate support advanced users can customize connection statements through `DatabaseUrl`. I will remove them first, the added fields have nothing to do with this pr. I'm going to complete it in next pr.

**Review by hsluoyz — APPROVED** [review]

**Pull request merged**